plugins/ledger/auth2fa.py

```diff
@@ -149,205 +124,32 @@ def pin_changed(s):
             self.modes.addItem(_("Security Card Challenge"))
-            if not self.cfg['pair']:
-                self.modes.addItem(_("Mobile - Not paired"))
-            else:
-                self.modes.addItem(_("Mobile - {}").format(self.cfg['pair'][1]))
         self.modes.blockSignals(False)
 
     def update_dlg(self):
         self.modes.setCurrentIndex(self.cfg['mode'])
         self.modebox.setVisible(True)
-        self.addPair.setText(_("Pair") if not self.cfg['pair'] else _("Re-Pair"))
-        self.addPair.setVisible(self.txdata['confirmationType'] > 2)
-        self.helpmsg.setText(helpTxt[self.cfg['mode'] if self.cfg['mode'] < 2 else 2 if self.cfg['pair'] else 4])
+        self.helpmsg.setText(helpTxt[self.cfg['mode']])
         self.helpmsg.setMinimumHeight(180 if self.txdata['confirmationType'] == 1 else 100)
-        self.pairbox.setVisible(False)
         self.helpmsg.setVisible(True)
         self.pinbox.setVisible(self.cfg['mode'] == 0)
         self.cardbox.setVisible(self.cfg['mode'] == 1)
         self.pintxt.setFocus(True) if self.cfg['mode'] == 0 else self.cardtxt.setFocus(True)
         self.setMaximumHeight(200)
 
-    def do_pairing(self):
-        rng = os.urandom(16)
-        pairID = (hexlify(rng) + hexlify(hashlib.sha256(rng).digest()[0:1])).decode('utf-8')
-        self.pairqr.setData(pairID)
-        self.modebox.setVisible(False)
-        self.helpmsg.setVisible(False)
-        self.pinbox.setVisible(False)
-        self.cardbox.setVisible(False)
-        self.pairbox.setVisible(True)
-        self.pairqr.setMinimumSize(300,300)
-        if self.ws:
-            self.ws.stop()
-        self.ws = LedgerWebSocket(self, pairID)
-        self.ws.pairing_done.connect(self.pairing_done)
-        self.ws.start()
-
-    def pairing_done(self, data):
-        if data is not None:
-            self.cfg['pair'] = [ data['pairid'], data['name'], data['platform'] ]
-            self.cfg['mode'] = 2
-            self.handler.win.wallet.get_keystore().cfg = self.cfg
-            self.handler.win.wallet.save_keystore()
-        self.pin = 'paired'
-        self.accept()
-
-    def req_validation(self):
-        if self.cfg['pair'] and 'secureScreenData' in self.txdata:
-            if self.ws:
-                self.ws.stop()
-            self.ws = LedgerWebSocket(self, self.cfg['pair'][0], self.txdata)
-            self.ws.req_updated.connect(self.req_updated)
-            self.ws.start()
-
-    def req_updated(self, pin):
-        if pin == 'accepted':
-            self.helpmsg.setText(helpTxt[3])
-        else:
-            self.pin = str(pin)
-            self.accept()
-
     def getDevice2FAMode(self):
         apdu = [0xe0, 0x24, 0x01, 0x00, 0x00, 0x01] # get 2fa mode
         try:
             mode = self.dongle.exchange( bytearray(apdu) )
             return mode
         except BTChipException as e:
-            debug_msg('Device getMode Failed')
+            print_msg('Device getMode Failed')
```

Comment:
Where is this symbol coming from?  I don't see it from imports.

`electroncash.util` has a `print_msg` but that is printing to stdout unconditionally.  Usually we do `print_error` or we inherit from `util.PrintError` and do `self.print_error`...

Reply:
Entirely your call. ha. :)

Reply:
Yea, I'll switch it to `print_error`. Also looks like I still need to import `BTChipException`.